kratos/processes/apply_periodic_boundary_condition_process.cpp

```diff
@@ -226,7 +226,7 @@ void ApplyPeriodicConditionProcess::ConstraintSlaveNodeWithConditionForVectorVar
             mrMasterModelPart.AddMasterSlaveConstraint(constraint6);
 
 
-            if (TDim == 3)
+            if (TDim == 3) // TODO: This function can be optimized using template specialization. if constexpr will fail due to unused variables declared
```

Comment:
Either way we should decide on going the constexp way or specializing, but both at the same time are ortogonal.